lib/shopify-cli/js_deps.rb

```diff
@@ -74,10 +77,19 @@ def npm(verbose = false)
       deps = %w(dependencies devDependencies).map do |key|
         pkg.fetch(key, []).keys
       end.flatten
+
+      success = true
       CLI::UI::Spinner.spin(ctx.message('core.js_deps.npm_installing_deps', deps.size)) do |spinner|
-        ctx.system(*cmd, chdir: ctx.root)
-        spinner.update_title(ctx.message('core.js_deps.npm_installed_deps', deps.size))
+        success = ctx.system(*cmd, chdir: ctx.root).success?
+
+        if success
+          spinner.update_title(ctx.message('core.js_deps.npm_installed_deps', deps.size))
+        else
+          spinner.update_title(ctx.message('core.js_deps.error.install_error'))
+        end
```

Comment:
I see you're trying to get around a block scoping problem here. initializing to `true` feels wrong but both of the other options I could think of are also kind of icky

1. initialize to nil (it's not true yet)
or 
2. return out of the block (optionally using a diff var name?) 

```ruby
success = CLI::UI::Spinner.spin(...) do |spinner|
  success =  ...

  if success 
    ...
   true
  else
    ....
    false
  end
end
success
```

idk I don't love either of these either

Reply:
Again it is a problem of the spinner. It doesn't return the yield from the inner block while the `Frame.open` call does. 

The spinner comes from an external library that has been copied to the `vendor` folder within the CLI. I feel it would be an upstream change to this package https://github.com/Shopify/cli-kit and then we would have to pull that change to the CLI. I am not sure how often the CLI updated the vendor packages as they are currently a bit out of date.

Reply:
thanks for the info. def out of scope, yeah